### PR TITLE
[MXNET-460] Ssd camera demo

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
This uses a webcam or video capture device to run SSD in real time.

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by larroy** [comment]
@zhreshold

**Comment by CodingCat** [comment]
Hi, the community has passed to vote about associating the code changes with JIRA (https://lists.apache.org/thread.html/ab22cf0e35f1bce2c3bf3bec2bc5b85a9583a3fe7fd56ba1bbade55f@%3Cdev.mxnet.apache.org%3E)

We have updated the guidelines for contributors in https://cwiki.apache.org/confluence/display/MXNET/Development+Process, please ensure that you have created a JIRA at https://issues.apache.org/jira/projects/MXNET/issues/ to describe your work in this pull request and include the JIRA title in your PR as [MXNET-xxxx] your title where MXNET-xxxx is the JIRA id

Thanks!

File: example/ssd/dataset/cv2Iterator.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import mxnet as mx
+import numpy as np
+import cv2
```
Comment (piiswrong) [inline example/ssd/dataset/cv2Iterator.py]: user may not have cv2. Use a try block?

Comment (larroy) [reply example/ssd/dataset/cv2Iterator.py]: There's other usages of cv2, such as in dataset/iterator  cv2 is required for this whole thing to work as it is.

**Comment by larroy** [comment]
Can we merge this? Or what does it need to be changed? @zhreshold  @piiswrong

File: example/ssd/demo.py [hunk 2]
```diff
@@ -131,13 +138,59 @@ def parse_data_shape(data_shape_str):
...
+    cv2.putText(frame, draw_text, tp0, cv2.FONT_HERSHEY_COMPLEX_SMALL, 0.5, (0,0,255))
 
+
+def network_path(prefix, network, data_shape):
+    return "{}{}_{}".format(prefix, network, data_shape)
+
+def run_camera(args,ctx):
+    assert args.batch_size == 1, "only batch size of 1 is supported"
+    logging.info("Detection threshold is {}".format(args.thresh))
+    iter = CameraIterator()
```
Comment (zhreshold) [inline example/ssd/demo.py]: iter is reserved keyword

File: example/ssd/demo.py [hunk 3]
```diff
@@ -131,13 +138,59 @@ def parse_data_shape(data_shape_str):
         raise ValueError("Unexpected data_shape: %s", data_shape_str)
     return data_shape
 
-if __name__ == '__main__':
-    args = parse_args()
-    if args.cpu:
-        ctx = mx.cpu()
-    else:
-        ctx = mx.gpu(args.gpu_id)
+def draw_detection(frame, det, class_names):
```
Comment (zhreshold) [inline example/ssd/demo.py]: now we have two versions of drawing function, one with cv2, one with matplotlib. Is it necessary?

File: example/ssd/demo.py [hunk 4]
```diff
@@ -156,3 +209,22 @@ def parse_data_shape(data_shape_str):
...
+        ctx = mx.gpu(args.gpu_id)
+
+    if args.camera:
+        run_camera(args, ctx)
+    else:
+        run_images(args, ctx)
+    return 0
+
+if __name__ == '__main__':
+    sys.exit(main())
```
Comment (zhreshold) [inline example/ssd/demo.py]: why sys.exit ?

**Comment by zhreshold** [comment]
@larroy Do you have a tiny demo video and put it in the README, otherwise people won't even notice it and don't know how to use it

**Comment by szha** [comment]
ping

**Comment by zhreshold** [comment]
Waiting for a README or small tutorial.

**Comment by larroy** [comment]
Added  info to the README

**Pull request merged**